homeassistant/components/dynalite/cover.py

```diff
@@ -0,0 +1,94 @@
+"""Support for the Dynalite channels as covers."""
+from typing import Callable
+
+from homeassistant.components.cover import CoverDevice
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+
+from .dynalitebase import DynaliteBase, async_setup_entry_base
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities: Callable
+) -> None:
+    """Record the async_add_entities function to add them later when received from Dynalite."""
+
+    @callback
+    def cover_from_device(device, bridge):
+        if device.has_tilt:
+            return DynaliteCoverWithTilt(device, bridge)
+        return DynaliteCover(device, bridge)
+
+    async_setup_entry_base(
+        hass, config_entry, async_add_entities, "cover", cover_from_device
+    )
+
+
+class DynaliteCover(DynaliteBase, CoverDevice):
+    """Representation of a Dynalite Channel as a Home Assistant Cover."""
+
+    @property
+    def device_class(self) -> str:
+        """Return the class of the device."""
+        return self._device.device_class
```

Comment:
can you please explain? not sure what you mean by the mapping

Reply:
We need to decouple the two APIs.
```py
DYNALITE_HA_DEVICE_CLASS = {
    "dyn_garage": "garage",
    ...
}
```

Reply:
this is true for the one in test_const. I can fix this. will comment on it below on the specific comment

for the device_class, this is simply the string that is passed in the config to the timecover. The library doesn't do anything specific with it and doesn't use any of the HA conf values, but it lets the HA user specify what class that specific cover should belong to (e.g. for UI reasons). There are no corresponding DEVICE_CLASS_XXX in the dynalite library and if we change them in HA, it won't require any change in the library or in the integration and will simply work, so I don't believe they should be considered coupled. However, if there is a more standard way to accomplish it, let me know and i will implement

Reply:
`device_class` here is part of the entity api and we're not allowed to return something that is not defined in Home Assistant. We should return one of the cover device class constants.

Reply:
now it is forced to return something from Home Assistant. Using the DEVICE_SCHEMA.
As far as the library is concerned, this is just a value that is stored and is returned as is from the config. The reason I have this in the library instead of in the component is that for the user, it is defined as an area for the cover (with the 3 presets/1 channel). However, the integration doesn't look into the specific devices to map them to the values in configuration.yaml since unfortunately, this is not a straightforward mapping, so I believe it is better to use this (similar to callback functions, where they can be passed without the library knowing their content).
Would the following change work? It will now always be a value from HA and if HA changes the list of values in components/cover/__init__.py, it wouldn't require any change to the integration so it doesn't create a true coupling

Reply:
Ok.

Reply:
But we should still have a map. Otherwise remove it from the library if it only passes it through.

Reply:
it only passes through, so i will try to remove it from the library
problem in configuring it directly is that it is not trivial to map from the devices to the relevant place in the config. Is there some example who does something similar? I was thinking about implementing the device_class in the integration and just taking the value from the right place in configuration.yaml, but trying to figure out how to map it.
Alternatively, I thought about just removing this function. If I do it, the user can still set the device class via the customizations, right? If so, no functionality is really lost. Do you think this is better since I cannot think of a solution that won't complicate the integration code and not sure the value is worth it, but you know HA better than me

Reply:
If removing is complicated you can implement the decoupling by looking up the possible home assistant device classes.
```py
device_class = next(
    (
        dev_class for dev_class in DEVICE_CLASSES
        if dev_class == self._device.device_class
    ),
    None
)
return device_class
```

Reply:
that makes it simpler. perhaps simpler with the same effect will be:
if self._device.device_class in DEVICE_CLASSES:
    return self._device.device_class
return DEFAULT_DEVICE_CLASS
is it the same effect?